providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java

```diff
@@ -247,53 +264,49 @@ private <T> ProviderEvaluation<T> getEvaluation(
                 errorCode = ErrorCode.GENERAL;
             }
             return ProviderEvaluation.<T>builder()
-                .errorCode(errorCode)
-                .reason(errorCode.name())
-                .value(defaultValue)
-                .build();
+                    .errorCode(errorCode)
+                    .reason(errorCode.name())
+                    .value(defaultValue)
+                    .build();
         }
-        ProviderEvaluation<T> res;
+
         GoFeatureFlagUser user = GoFeatureFlagUser.fromEvaluationContext(evaluationContext);
         if (cache == null) {
+            // Cache is disabled we return directly the remote evaluation
             EvaluationResponse<T> proxyRes = resolveEvaluationGoFeatureFlagProxy(key, defaultValue, user, expectedType);
-            res = proxyRes.getProviderEvaluation();
-        } else {
-            res = getProviderEvaluationWithCheckCache(key, defaultValue, expectedType, user);
-            eventsPublisher.add(Event.builder()
-                .key(key)
-                .defaultValue(defaultValue)
-                .variation(res.getVariant())
-                .value(res.getValue())
-                .userKey(user.getKey())
-                .creationDate(System.currentTimeMillis())
-                .build()
-            );
+            return proxyRes.getProviderEvaluation();
         }
-        return res;
-    }
 
-    private <T> ProviderEvaluation getProviderEvaluationWithCheckCache(
-            String key, T defaultValue, Class<?> expectedType, GoFeatureFlagUser user) {
-        ProviderEvaluation<?> res;
+        String cacheKey = null;
         try {
-            String cacheKey = buildCacheKey(key, BeanUtils.buildKey(user));
-            res = cache.getIfPresent(cacheKey);
-            if (res == null) {
-                EvaluationResponse<T> proxyRes = resolveEvaluationGoFeatureFlagProxy(
-                    key, defaultValue, user, expectedType);
-                if (Boolean.TRUE.equals(proxyRes.getCachable())) {
-                    cache.put(cacheKey, proxyRes.getProviderEvaluation());
-                }
-                res = proxyRes.getProviderEvaluation();
-            } else {
-                res.setReason(CACHED_REASON);
+            cacheKey = buildCacheKey(key, BeanUtils.buildKey(user));
+            ProviderEvaluation<?> cacheResponse = cache.getIfPresent(cacheKey);
+            if (cacheResponse == null) {
+                throw new NotPresentInCache(cacheKey);
+            }
+            cacheResponse.setReason(CACHED_REASON);
+            eventsPublisher.add(Event.builder()
```

Comment:
why skip the event when value is evaluated from proxy ?

Reply:
see https://github.com/open-feature/java-sdk-contrib/pull/369#discussion_r1284440475